numpy/core/setup.py

```diff
@@ -605,6 +606,12 @@ def generate_api(ext, build_dir):
     config.add_include_dirs(join('src', 'umath'))
     config.add_include_dirs(join('src', 'npysort'))
 
+    # Need to add location of npy_math_internal.h, generated from src/npymath/npy_math_internal.h.src
+    npymath_build_dir = join('build', 'src.%s-%s.%s' % (sysconfig.get_platform(), sys.version_info[0], sys.version_info[1]),
+                             'numpy', 'core', 'src', 'npymath')
+    config.add_include_dirs(npymath_build_dir)
+    config.add_define_macros([("NPY_BUILD", "1")]) # this macro indicates that Numpy build is in process
```

Comment:
Ah true. For the sake of those reading through the code, the latter looks like a better option.